### PR TITLE
Commenting pagination

### PR DESCRIPTION
**Pull request by kathtran**
Please let me know if I should add more comments or if any of the existing comments should be modified to better communicate how pagination operates.

### CODE REVIEW

**Comment by RickyV33** [comment]
LGTM

**Comment by chances** [comment]
Can you describe the parameters?

e.g. ` @param maxResults Maximum number of items to collect in the array of response data`

Also, you said "collecting any interval of pages," but the function doesn't do that. It collects a set of paged data beginning at a given page. Collection continues until either the end of the paged data is reached or the number of items collected is equal to the given number of maximum results, whichever comes first.

File: lib/pagination.js [hunk 1]
```diff
@@ -3,6 +3,18 @@ let request = require('request');
...
 const PAGE_SIZE = 20;
 
+/**
+ * Requests to the Jama API are paginated, so this function collects a set of paged data, beginning at a
+ * given page from a given endpoint into an array of response data. Collection continues until either the
+ * end of the paged data is reached or the number of items collected is equal to a given number of maximum
+ * results, whichever comes first.
+ *
+ * @param baseUrl Endpoint that data is pulled from
+ * @param startAt Numeric value used to mark the starting point of each page (interval is set to 20)
```
Comment (chances) [inline lib/pagination.js]: This isn't a starting point of each page. Isn't this the index within the Jama data to start at? So, like if I've already gotten the first 30 items, I could pass 30 to this function to get items starting at the 30th index, right?

File: lib/pagination.js [hunk 2]
```diff
@@ -3,6 +3,18 @@ let request = require('request');
...
+ * @param maxResults Maximum number of items to collect in the array of response data
+ * @returns {Promise} If resolved, the payload that contains all data from the pages is returned. Else, an
+ * error object is returned.
```
Comment (chances) [inline lib/pagination.js]: Otherwise, the promise is rejected with an error object.

File: lib/pagination.js [hunk 3]
```diff
@@ -3,6 +3,18 @@ let request = require('request');
...
+/**
+ * Requests to the Jama API are paginated, so this function collects a set of paged data, beginning at a
+ * given page from a given endpoint into an array of response data. Collection continues until either the
+ * end of the paged data is reached or the number of items collected is equal to a given number of maximum
+ * results, whichever comes first.
+ *
+ * @param baseUrl Endpoint that data is pulled from
+ * @param startAt Numeric value used to mark the starting point of each page (interval is set to 20)
+ * @param maxResults Maximum number of items to collect in the array of response data
+ * @returns {Promise} If resolved, the payload that contains all data from the pages is returned. Else, an
```
Comment (chances) [inline lib/pagination.js]: "payload" is an implementation detail. Shouldn't it be said that the promise resolves with an array of all the data?

File: lib/pagination.js (hunk 1 above)
Comment (kathtran) [inline lib/pagination.js]: I had a hard time phrasing it. But we have it hard coded to start at zero right now, so what should I do? @chances

Comment (chances) [inline lib/pagination.js]: "Index within span of the Jama collection to start at (e.g To start at the tenth item, pass 10)"

Or similar?

**Pull request merged**